grails-app/services/org/pih/warehouse/requisition/RequisitionService.groovy

```diff
@@ -860,9 +859,15 @@ class RequisitionService {
                 if (!requisition.origin.approvalRequired) {
                     throw new IllegalArgumentException("Fulfilling location must support Request Approval")
                 }
-                transitionRequisitionStatus(requisition, RequisitionStatus.REJECTED, EventCode.REJECTED, currentUser)
+                requisition.status = RequisitionStatus.REJECTED
                 requisition.dateRejected = new Date()
                 requisition.rejectedBy = currentUser
+
+                Event event = createEvent(EventCode.REJECTED, requisition.origin, new Date(), currentUser)
+                event.comment = comment
+                requisition.addToEvents(event)
+
+                requisition.addToComments(comment)
```

Comment:
Instead of getting rid of calling the `transitionRequisitionStatus` here, I'd suggest refactoring the `transitionRequisitionStatus` method to include this part, so to add there something like:
```groovy
if (comment) {
  event.comment = comment
  requisition.addToEvents(event)
  requisition.addToComments(comment)
}
```
By the way it feels a bit dangerous to me, that we use the same comment instance adding it to a Requisition and to an Event - do we really need in both places? 
Assuming we'd want to delete a comment, we'd have to remember to check if a Comment is not also associated with an Event, because otherwise it would not allow us to delete it (some of the comments would, some not, that's why it makes me confused about this approach).
I will come back to it on Monday, because I feel like I miss some context.

Reply:
The reason I also ask about it, is that I don't see you "extracting" a Comment from an Event anywhere (in the view, or anywhere on the backend side) - I see you only adding it, so I don't know what's the purpose of making a relationship between an Event and a Comment.

Reply:
I agree this could be in the `transitionRequisitionStatus`. It could have a null by default comment parameter.